snf-cyclades-app/synnefo/api/util.py

```diff
@@ -188,6 +189,13 @@ def get_image(image_id, user_id):
             raise faults.ItemNotFound("Image '%s' not found" % image_id)
 
 
+def get_keypair(keypair_name, user_id):
+    try:
+        return PublicKeyPair.objects.get(name=keypair_name, user=user_id)
+    except Exception as e:
```

Comment:
We shouldn't catch (and re-raise) generic exceptions here. Use the pattern with `faults.ItemNotFound` as in other functions.